### PR TITLE
Update `SubmissionProxy`

### PR DESCRIPTION
**Pull request by martinkersner**
# Description

* Removed `contracts/v0.1/src/BatchSubmission.sol`
* Unify solidity version (0.8.24) in v0.2.
* Unify licenses -> everything MIT.
* Applied `forge fmt`.
* Every oracle registered to `SubmissionOracle` is limited by expiration period (`expirationPeriod`). After expiration period ends, oracle cannot submit values anymore.
* Convert all `public` function in `SubmissionProxy` to `external`.
* Tests.

## Type of change

Please delete options that are not relevant.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist before requesting a review

- [x] I have performed a self-review of my code.
- [x] If it is a core feature, I have added thorough tests.

## Summary by CodeRabbit

- **New Features**
	- Introduced functionality for setting submission expiration periods and managing oracles with expiration times in the SubmissionProxy contract.
- **Enhancements**
	- Updated Solidity version across various contracts for compatibility and performance improvements.
	- Improved readability and clarity in code formatting and logic across multiple contracts.
	- Streamlined event declarations and function signatures for better usability.
- **Bug Fixes**
	- Adjusted loop variable types and simplified conditional statements for more reliable execution.
- **Documentation**
	- Updated SPDX license identifiers to MIT in applicable contracts.

### CODE REVIEW

**Comment by martinkersner** [comment]
@coderabbitai generate interesting stats about this repository and render them as a table.

**Comment by martinkersner** [comment]
> Tip
> 
> For best results, initiate chat on the files or code changes.
> 
> @martinkersner: Here are the interesting stats about the repository, rendered as a table:
> 
> ```
> | Statistic         | Value  |
> |-------------------|--------|
> | Number of Authors | 13     |
> | Lines of Code     | 226,254|
> | TODO Comments     | 16     |
> ```
> 
> These statistics provide a snapshot of the repository's current state, including its collaborative nature, codebase size, and areas potentially marked for future work or improvements.
> 
> Analysis chain
> Scripts executed

I want to know actual number lines of code. You seem to be counting yarn.lock as well. Please exclude it.

File: contracts/v0.2/src/SubmissionProxy.sol [hunk 1]
```diff
@@ -1,65 +1,103 @@
...
+            revert InvalidSubmissionLength();
+        }
 
-    function batchSubmit(
-        address[] memory _aggregators,
-        int256[] memory _submissions
-    ) public onlyOracle {
-        if (_aggregators.length != _submissions.length) revert InvalidSubmissionLength();
-        for (uint256 i = 0; i < _aggregators.length; i++) {
+        for (uint256 i = 0; i < _aggregators.length; ++i) {
```
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/src/SubmissionProxy.sol:100]: TODOs are meant for next PRs. Suggest me how to implement a validation for the setMaxSubmission and setExpirationPeriod.

File: contracts/v0.2/deploy/Utils.s.sol [hunk 2]
```diff
@@ -44,18 +47,13 @@ contract UtilsScript is Script {
...
             bool fileExisted = false;
             migrationFileName = vm.readLine(lockFilePath);
             migrationFilePath = string.concat(path, "/", migrationFileName);
 
             while (bytes(migrationFileName).length > 0) {
-                if (
-                    keccak256(abi.encodePacked(migrationFilePath)) ==
-                    keccak256(abi.encodePacked(entry.path))
-                ) {
+                if (keccak256(abi.encodePacked(migrationFilePath)) == keccak256(abi.encodePacked(entry.path))) {
```
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/deploy/Utils.s.sol:56]: Just updated formatting. Don't care about it now.

File: contracts/v0.2/test/SubmissionProxy.t.sol [hunk 3]
```diff
@@ -1,16 +1,16 @@
...
-pragma solidity ^0.8.13;
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
 
-import {Test, console2, console} from "forge-std/Test.sol";
+import {Test, console} from "forge-std/Test.sol";
 import {SubmissionProxy} from "../src/SubmissionProxy.sol";
 import {Aggregator} from "../src/Aggregator.sol";
 
+// TODO test submit after oracle expires
```
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/test/SubmissionProxy.t.sol:8]: It is included already. I am going to remove it.

File: contracts/v0.2/src/SubmissionProxy.sol (hunk 1 above)
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/src/SubmissionProxy.sol:100]: Not bad! Thanks!

File: contracts/v0.2/src/SubmissionProxy.sol [hunk 4]
```diff
@@ -1,65 +1,111 @@
...
+    error InvalidOracle();
+    error InvalidSubmissionLength();
+    error InvalidExpirationPeriod();
+    error InvalidMaxSubmission();
+
+    modifier onlyOracle() {
+        uint256 expiration = expirations[msg.sender];
+        if (expiration == 0 || expiration < block.timestamp) revert OnlyOracle();
+        _;
+    }
```
[the remark this answers is not in the document]
Comment (martinkersner) [reply contracts/v0.2/src/SubmissionProxy.sol:31]: You suggestion is actually less restrictive than the current implementation. Overall, it is not a big deal.

**Comment by nick-bisonai** [comment]
is batch submission based on pair names strings rather than aggregator addresses to be implemented later?

**Comment by martinkersner** [comment]
> is batch submission based on pair names strings rather than aggregator addresses to be implemented later?

Yes!

**Review by nick-bisonai — APPROVED** [review]
lgtm!

File: contracts/v0.2/src/SubmissionProxy.sol [hunk 5]
```diff
@@ -7,14 +7,18 @@ import {IAggregator} from "./interfaces/IAggregatorSubmit.sol";
...
+    uint256 public constant MIN_EXPIRATION = 1 days;
+    uint256 public constant MAX_EXPIRATION = 365 days;
+
     uint256 public maxSubmission = 50;
     uint256 public expirationPeriod = 5 weeks;
 
-    address[] public oracles;
-    mapping(address => uint256) expirations;
+    // oracle => expiration
+    mapping(address => uint256) oracles;
```
Comment (nick-bisonai) [inline contracts/v0.2/src/SubmissionProxy.sol:19]: it's optional, how about removing `// oracle => expiration` comment and updating into 
```
mapping(address oracleAddress => uint256 expiration) oracles
```
solidity supports named mapping
reference: https://soliditylang.org/blog/2023/02/01/solidity-0.8.18-release-announcement/

Comment (martinkersner) [reply contracts/v0.2/src/SubmissionProxy.sol]: Looks great!

**Pull request merged**